### DISCUSSION TITLE
StateCaster only checks for null, not empty strings

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/laravel-model-states#182] [Bugs]
I wanted to add states to a model that didn't have any. I created the migration, added the new column to `$casts` and `$attributes`. This is it. Based on the title, you can probably guess what I have missed. Yes, I forgot to add values to the models that were already there.

Of course our tests failed, rightly so. In the `StateCaster`'s `get()` method you check if the value is `null`, not if it's an empty string. Which I have just found, can happen too.

Although I classed this discussion as a bug, there might be a good reason to leave it as is. When I debugged why our tests failed, this did point me in the right direction. If you do think this is a bug though, let me know and I would be happy to raise a PR.

ps. thanks for all of your work. It is just insane how useful your packages are. Thanks, thanks, thanks!

**Comment by brendt** [comment]
I don't think we want to support this. It's _technically_ possible to have an empty string as a state value, and disallowing so would require a bunch of checks here and there.

**Reply by ju5t** [reply]
I didn't know that. I can't judge if this is something to support or not, so we should probably leave it as is.

**Reply by brendt** [reply]
No problem, it's good to share these ideas!
